src/classes/HiFiMixerSession.ts

```diff
@@ -831,26 +836,15 @@ export class HiFiMixerSession {
     }
 
     /**
-     * Sets the state of the current connection, and
-     * fires the onChange handler if that state has, in fact, changed.
+     * Fire the onChange handler for a state operation
      * @param state
      */
-    _setCurrentHiFiConnectionState(state: HiFiConnectionStates): void {
-        if (this._currentHiFiConnectionState !== state) {
-            this._currentHiFiConnectionState = state;
-            if (this.onConnectionStateChanged) {
-                this.onConnectionStateChanged(this._currentHiFiConnectionState);
-            }
+    _onConnectionStateChange(state: HiFiConnectionStates): void {
+        if (this.onConnectionStateChanged) {
+            this.onConnectionStateChanged(state);
         }
     }
 
-    /**
-     * Return the current state of the connection.
-     */
-    getCurrentHiFiConnectionState(): HiFiConnectionStates {
-        return this._currentHiFiConnectionState;
-    }
-
```

Comment:
I appreciate the effort to "clear the decks" and remove all the complications that we don't need.

But one reason for tracking the state (and making it available to applications), is so that an application can query the state at odd times. For example, consider the navigator property https://developer.mozilla.org/en-US/docs/Web/API/NavigatorOnLine/onLine -- Some apps want to know when this changes, and so there is an event that people can listen on. But sometimes, in a handler for something else entirely, one might want to know that we are offline, and it is convenient to just ask. Yes, an application could register for changes and keep track of the last change itself, but the more untested user code that needs to be written -- by each individual application writer -- the more chance that there will be bugs. The mark of a great API isn't that it has the minimum number of API bugs, but that all the applications that use it have the minimum number of bugs.

Reply:
This method didn't get removed, it got moved up to HiFiCommunicator rather than living on HiFiMixerSession. (Users can't actually access HiFiMixerSession directly.)

Reply:
Excellent. (Hadn't read that code yet.)